### PR TITLE
Update the root readme to include clear acquisition and installation details.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Things left to do (out-of-scope for this PR):

- Add FAQ section for common errors and potential solutions when trying to build via vcpkg
- Structure the doc with the following groups, if there is not a lot of overlap:
  - Windows
    - Visual Studio
    - VS Code
    - Command Line
  - Linux/Posix/Mac
    - VS Code
    - Command Line
 - Another dimension is `Vcpkg` vs `fetch content from GitHub` vs `build the SDK from source`.

### CODE REVIEW

File: README.md
Comment (antkmsft) [inline README.md:102]: Something wrong with indentation?

File: README.md
Comment (antkmsft) [inline README.md:107]: use `std::cout`, the "more C++ way"?

File: README.md
Comment (antkmsft) [inline README.md:105]: catch by const ref?

File: README.md
Comment (antkmsft) [inline README.md:160]:
```suggestion
> NOTE: In case of getting linker errors when consuming the SDK on Windows, make sure that [VcPkg triplet](https://vcpkg.readthedocs.io/en/latest/users/triplets/) being consumed matches the [CRT link flags](https://docs.microsoft.com/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-160) being set for your app or library build. See also `MSVC_USE_STATIC_CRT` build flag.

```

**Comment by ahsonkhan** [comment]
cc @kyle-patterson

File: README.md
Comment (ahsonkhan) [inline README.md:112]: I avoided including this in https://github.com/Azure/azure-sdk-for-cpp/tree/UpdatedReadMe since it reveals a bit too much that can just be copy/pasted.

File: README.md
Comment (antkmsft) [inline README.md:103]: If GetConnectionString() is not a part of SDK, but is something that user has to define, do the declaration after the include headers and before "usnig namespace": `std::string GetConnectionString();`

File: README.md
Comment (antkmsft) [inline README.md:37]: If we go to explain vcpkg basics, I'd specify, what the triplet is: "the triplet is a combination of a, b, and c [link to vcpkg docs]". And in this sentence, "optionally specifying the triplet (such as X, Y etc.)"

File: README.md
Comment (antkmsft) [inline README.md:43]:
```suggestion
For example, this will install packages for the `x64-windows` triplet. On Windows, not specifying a triplet will default to x86-windows:

```

Comment (antkmsft) [inline README.md:43]: "On Windows, not specifying a triplet will default to x86-windows" - are we sure?

File: README.md
Comment (antkmsft) [inline README.md:19]: This is great, but do we want to go to explanation of how VcPkg works? Do they have a doc that maybe we can link to instead?

File: README.md
Comment (antkmsft) [inline README.md:5]: Do we reed to say "public developer docs", or can we just say "developer docs"?

**Review by antkmsft — APPROVED** [review]

File: README.md
Comment (antkmsft) [reply README.md:103]: Ore, maybe, even put a body:
```
std::string GetConnectionString()
{
    return ...
}
```

File: README.md
Comment (RickWinter) [inline README.md:9]:
```suggestion
For the best development experience, we recommend developers use [CMake projects in Visual Studio](https://docs.microsoft.com/cpp/build/cmake-projects-in-visual-studio?view=vs-2019) to view and build the source code together with its dependencies. You can also use any other text editor of your choice, such as [VS Code](https://code.visualstudio.com/), along with the command line for building your application with the SDK.

```

File: README.md
Comment (RickWinter) [inline README.md:11]:
```suggestion
You can find additional information for specific libraries by navigating to the appropriate folder in the `/sdk` directory. See the **README.md** file located in the library's project folder. For example, [Azure Storage client library](https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/storage#azure-storage-client-library-for-c).

```

File: README.md
Comment (RickWinter) [inline README.md:13]:
```suggestion
For API reference docs, tutorials, samples, quick starts, and other documentation, go to [Azure SDK for C++ Developer Docs](https://azure.github.io/azure-sdk-for-cpp).

```

File: README.md
Comment (RickWinter) [inline README.md:17]:
```suggestion
The easiest way to acquire the C++ SDK is leveraging [vcpkg](https://github.com/microsoft/vcpkg). [Git](https://git-scm.com/downloads) is required before getting started.

```

File: README.md
Comment (RickWinter) [reply README.md:37]: I'd avoid too much detail and instead show "Install x64 on Windows" followed by steps.  Have 2 or 3 of the basic exact commands.  Then link to vcpkg for those that want a more in depth explanation.  
We should avoid being documentation for vcpkg.  Also you should wait to provide deep dive links to external sites as each time its an opportunity to "lose" them in the journey.

**Review by RickWinter — CHANGES_REQUESTED** [review]

File: README.md
Comment (ahsonkhan) [reply README.md:19]: They do, but having the end user link back-forth between various doc locations isn't a good experience. There are only ~3 steps or the customer to follow, so in-lining them here helps them get started easily. I have a link below to the vcpkg repo if they need more info.

File: README.md
Comment (ahsonkhan) [reply README.md:43]: Yes. :)

File: README.md
Comment (ahsonkhan) [reply README.md:160]: This was an intentional casing change to be consistent across the board, and with what the actual `Vcpkg` repo does. I didn't find this mixed casing used anywhere. For example, see https://github.com/microsoft/vcpkg#vcpkg-with-visual-studio-cmake-projects

File: README.md
Comment (ahsonkhan) [reply README.md:37]: Agreed, that was my approach with the docs here.

File: README.md
Comment (ahsonkhan) [reply README.md:11]: This is what you are suggesting here, which isn't really a sentence
> For example, Azure Storage client library.`

Maybe I can merge it in the previous sentence:
> See the README.md file located in the library's project folder, for example, the Azure Storage client library.

File: README.md
Comment (antkmsft) [reply README.md:160]: Tha's good, then please make sure we don't have occurrences of using "VcPkg" - I think we have paragraph title called that.

File: README.md
Comment (ahsonkhan) [reply README.md:17]: I had heard feedback in the past not to use a passive "reference manual" style tone (e.g. x is required) and refer to the reader directly in an active voice (e.g. you need x), which tends to be more engaging.

How about:
> You will need to install Git before getting started?

Comment (antkmsft) [inline README.md:17]:
```suggestion
The easiest way to get the C++ SDK is to leverage [Vcpkg](https://github.com/microsoft/vcpkg). You'll need to have [Git](https://git-scm.com/downloads) installed before getting started.

```

File: README.md
Comment (antkmsft) [inline README.md:19]:
```suggestion
First clone and bootstrap Vcpkg itself. You can install it anywhere on your machine, but **make note** of the directory where you clone the Vcpkg repo.

```

File: README.md
Comment (ahsonkhan) [reply README.md:160]: Can you be specific if there are any that I missed in this README? I only saw one and fixed it already.

File: README.md
Comment (antkmsft) [inline README.md:49]:
```suggestion
See the [list of packages](https://github.com/Azure/azure-sdk-for-cpp#vcpkg) available via Vcpkg below. All Azure C++ SDK package names start with `azure-`.

```

File: README.md
Comment (antkmsft) [inline README.md:70]:
```suggestion
> NOTE: All the Azure client libraries take a dependency on `azure-core-cpp` which provides functionality commonly needed by all Azure clients. When you install any client library via Vcpkg, it will bring in all the necessary dependencies as well. You don't need to install those individually to get started.

```

File: README.md
Comment (antkmsft) [inline README.md:72]:
```suggestion
You can reference this [Vcpkg Quick Start](https://github.com/microsoft/vcpkg#quick-start-windows) for more details.

```

File: README.md
Comment (antkmsft) [inline README.md:76]:
```suggestion
In order to use the SDK installed via Vcpkg with CMake, you can use the toolchain file from Vcpkg:

```

File: README.md
Comment (antkmsft) [inline README.md:79]:
```suggestion
> cmake -B [build directory] -S . -DCMAKE_TOOLCHAIN_FILE=[path to Vcpkg repo]/scripts/buildsystems/vcpkg.cmake

```

File: README.md
Comment (antkmsft) [inline README.md:122]:
```suggestion
        "cmakeToolchain": "<path to Vcpkg repo>/vcpkg/scripts/buildsystems/vcpkg.cmake",

```

**Review by antkmsft — COMMENTED** [review]
Let's also convert all other occurrences of "VcPkg" and some of "vcpkg" to "Vcpkg" in our entire repo.

File: README.md
Comment (ahsonkhan) [reply README.md:17]: Following existing vcpkg docs, I don't think we need to capitalize it if isn't in a title or start of a sentence. For example:
https://github.com/microsoft/vcpkg#vcpkg-as-a-submodule
https://vcpkg.readthedocs.io/en/latest/tool-maintainers/layout/
https://en.wikipedia.org/wiki/Vcpkg
https://docs.microsoft.com/en-us/cpp/build/vcpkg?view=msvc-160&viewFallbackFrom=vs-2019

File: README.md
Comment (ahsonkhan) [reply README.md:19]: Leaving as is, lower-case unless title or start of sentence.

File: README.md
Comment (antkmsft) [reply README.md:43]: I am not questioning you saw it, I meant would this be the case on all customer Windows machines, i.e. is this behavior guaranteed/documented, or is it selected based on some system variables, and/or the default be changed by Vcpkg in the future? Is it possible to rewrite as less certain statement/avoid the statement about Vcpkg defaults?

File: README.md
Comment (antkmsft) [reply README.md]: Oh, so it looks like the name is lowercase "vcpkg". Then sure, we don't capitalize it unless it starts the sentence. I think we need another set of case fixes then.

File: README.md
Comment (ahsonkhan) [inline README.md:17]: @RickWinter I removed the link to the vcpkg repo from here, since I link it later below and also removed example commands like `> .\vcpkg\vcpkg install [packages to install]`, leaving just the concrete ones.

Let me know if this is what you had in mind, and if you have any other suggestions.

File: README.md
Comment (antkmsft) [inline README.md:152]:
```suggestion
> NOTE: In case of getting linker errors when consuming the SDK on Windows, make sure that [vcpkg triplet](https://vcpkg.readthedocs.io/en/latest/users/triplets/) being consumed matches the [CRT link flags](https://docs.microsoft.com/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-160) being set for your app or library build. See also `MSVC_USE_STATIC_CRT` build flag.

```

File: README.md
Comment (ahsonkhan) [reply README.md]: Yes, I had chatted with vcpkg folks and they mentioned the default is x86 which won't be changed.
If you don't have any specific vcpkg environment variables set (i.e. clean Windows machine), yes, that is the default.

I think it is important to call this out as it can be surprising to folks since they get hard to debug errors when trying to find the package and build their application. I don't see much benefit to try to make a less certain statement.

Here's a doc that mentions it:
https://vcpkg.readthedocs.io/en/latest/users/triplets/#additional-remarks

File: README.md
Comment (vhvb1989) [inline README.md:85]: Well, this gives the impression that we are recommending having the connection string in the source code, which is bad.

Since this is an example of how we would do a sample app, can we use `std::getenv()` to recommend always putting connection string on the env?

File: README.md
Comment (vhvb1989) [inline README.md:77]: We can remove this if we are not using anything from core.hpp directly.
If you want to keep it, add something like a body stream or context

File: README.md
Comment (vhvb1989) [inline README.md:108]: I would be surprised by people working with the json directly instead of the json viewer from Visual Studio. ( No actions for this comment, just sharing with you)

**Review by vhvb1989 — APPROVED** [review]

File: README.md
Comment (ahsonkhan) [reply README.md:77]: `RequestFailedException` gets used.

File: README.md
Comment (ahsonkhan) [reply README.md:108]: I was considering including a picture like this, later:
image

File: README.md
Comment (ahsonkhan) [reply README.md:85]: This isn't really the point of the sample, and I didn't want to encourage use of `getenv()` especially without "is string empty" checks. Doing it properly ends up taking up too much code space. But you bring up a good point.

Maybe I should just leave it without a body and add a comment that the user can read it from an environment variable.

**Comment by ahsonkhan** [comment]
@msftbot do not merge unless @RickWinter approves

**Comment by ahsonkhan** [comment]
@msftbot hold this pr for the next 1 hour

**Comment by bhardwajs** [comment]
@ahsonkhan Is consuming SDK as a submodule supported scenario?

**Comment by ahsonkhan** [comment]
> Is consuming SDK as a submodule supported scenario?

We plan it to be, and it might just work already, but I haven't tried it yet. It is a bit lower in the priority list. Do you know of any particular pitfalls to look out for or document that is unique to consuming as a submodule (which would be different from just building the SDK as source outside of the application repo)?

**Comment by RickWinter** [comment]
> @ahsonkhan Is consuming SDK as a submodule supported scenario?

Consuming as a submodule is not the mainline consumption scenario.  It may work but is not officially supported.  
The reason for that is we do not expect all packages to ship on the same cadence.  i.e. Storage may ship 1.0 at the same time KeyVault ships the first Beta.  The folders within sdk directory are only verified stable at the release tags for each package (i.e. sdk/<folder>)

File: README.md
Comment (RickWinter) [reply README.md:17]: Seems reasonable.  We should give enough detail to onboard and unblock the customer while not becoming the vcpkg documentation.

**Comment by ahsonkhan** [comment]
> The reason for that is we do not expect all packages to ship on the same cadence.

Ah yes, good reminder Rick. Even if you submodule for a particular GA tag, it may contain libraries that are still in beta.

**Review by RickWinter — APPROVED** [review]

**Pull request merged**
